tests/components/fail2ban/test_sensor.py

```diff
@@ -60,158 +56,140 @@ def fake_log(log_key):
     return fake_log_dict[log_key]
 
 
-class TestBanSensor(unittest.TestCase):
-    """Test the fail2ban sensor."""
-
-    def setUp(self):
-        """Set up things to be run when tests are started."""
-        self.hass = get_test_home_assistant()
-        self.addCleanup(self.hass.stop)
-
-    @patch("os.path.isfile", Mock(return_value=True))
-    def test_setup(self):
-        """Test that sensor can be setup."""
-        config = {"sensor": {"platform": "fail2ban", "jails": ["jail_one"]}}
-        mock_fh = MockOpen()
-        with patch(
-            "homeassistant.components.fail2ban.sensor.open", mock_fh, create=True
-        ):
-            assert setup_component(self.hass, "sensor", config)
-            self.hass.block_till_done()
-        assert_setup_component(1, "sensor")
-
-    @patch("os.path.isfile", Mock(return_value=True))
-    def test_multi_jails(self):
-        """Test that multiple jails can be set up as sensors.."""
-        config = {"sensor": {"platform": "fail2ban", "jails": ["jail_one", "jail_two"]}}
-        mock_fh = MockOpen()
-        with patch(
-            "homeassistant.components.fail2ban.sensor.open", mock_fh, create=True
-        ):
-            assert setup_component(self.hass, "sensor", config)
-            self.hass.block_till_done()
-        assert_setup_component(2, "sensor")
-
-    def test_single_ban(self):
-        """Test that log is parsed correctly for single ban."""
-        log_parser = BanLogParser("/test/fail2ban.log")
-        sensor = BanSensor("fail2ban", "jail_one", log_parser)
-        assert sensor.name == "fail2ban jail_one"
-        mock_fh = MockOpen(read_data=fake_log("single_ban"))
-        with patch(
-            "homeassistant.components.fail2ban.sensor.open", mock_fh, create=True
-        ):
-            sensor.update()
-
+@patch("os.path.isfile", Mock(return_value=True))
+async def test_setup(hass):
+    """Test that sensor can be setup."""
+    config = {"sensor": {"platform": "fail2ban", "jails": ["jail_one"]}}
+    mock_fh = mock_open()
+    with patch("homeassistant.components.fail2ban.sensor.open", mock_fh, create=True):
+        assert await async_setup_component(hass, "sensor", config)
```

Comment:
Codacy found an issue: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/foreign-sub/home-assistant/pullRequest?prid=6333224)